lib/matplotlib/tests/test_mathtext.py

```diff
@@ -510,3 +510,12 @@ def test_mathtext_cmr10_minus_sign():
     ax.plot(range(-1, 1), range(-1, 1))
     # draw to make sure we have no warnings
     fig.canvas.draw()
+
+
+@check_figures_equal(extensions=["png"])
+def test_boldsymbol(fig_test, fig_ref):
+    fig_test.text(0.1, 0.1, r"$\boldsymbol{abc0123\alpha}")
+    fig_test.text(0.1, 0.2, r"$\boldsymbol{\mathrm{abc0123\alpha}}")
+
+    fig_ref.text(0.1, 0.1, r"$\boldsymbol{abc0123\alpha}")
+    fig_ref.text(0.1, 0.2, r"$\boldsymbol{\mathrm{abc0123\alpha}}")
```

Comment:
```suggestion
    fig_ref.text(0.1, 0.2, r"$\mathrm{abc0123\alpha}")
```
? (Otherwise the code for the two figures are identical.)

However, I think that you can simply add a string like this to the other test.

Reply:
I added the `\boldsymbol{\mathrm{string}}` to add a check for nested commands, to check that the innermost is taking precedence. Should I update it to test `\mathrm{string}`?

Reply:
I think it is enough to have the image test. But on the other hand, no harm in having the figures-equal test as well. (And they test slightly different things. For example, if boldsymbol was adding spaces, it would be detected by the figures-equal test, but not the image test.)